core/src/main/scala/org/bitcoins/core/script/crypto/CryptoInterpreter.scala

```diff
@@ -81,60 +81,69 @@ sealed abstract class CryptoInterpreter {
       val flags = program.flags
       val restOfStack = program.stack.tail.tail
 
-      program.txSignatureComponent.sigVersion match {
-        case SigVersionWitnessV0 | SigVersionWitnessV0 | SigVersionBase =>
-          val pubKey = ECPublicKeyBytes(program.stack.head.bytes)
-          val signature = ECDigitalSignature(program.stack.tail.head.bytes)
-          val removedOpCodeSeparatorsScript =
-            BitcoinScriptUtil.removeOpCodeSeparator(program)
-          val result = TransactionSignatureChecker.checkSignature(
-            program.txSignatureComponent,
-            removedOpCodeSeparatorsScript,
-            pubKey,
-            signature,
-            flags)
-          handleSignatureValidation(program = program,
-                                    result = result,
-                                    restOfStack = restOfStack,
-                                    numOpt = None)
-        case SigVersionTapscript =>
-          val tapscriptE: Either[
-            ScriptError,
-            TransactionSignatureCheckerResult] = evalChecksigTapscript(program)
-          tapscriptE match {
-            case Left(err) =>
-              if (
-                err == ScriptErrorDiscourageUpgradablePubkeyType && !ScriptFlagUtil
-                  .discourageUpgradablePublicKey(flags)
-              ) {
-                //trivially pass signature validation as required by BIP342
-                //when the public key type is not known and the
-                //SCRIPT_VERIFY_DISCOURAGE_UPGRADABLE_PUBKEYTYPE is NOT set
-                handleSignatureValidation(program = program,
-                                          result = SignatureValidationSuccess,
+      val sigBytes = program.stack.tail.head.bytes
+
+      if (
+        sigBytes.nonEmpty && program.validationWeightRemaining.forall(_ >= 50)
```

Comment:
`.forall(_ > 0)`

Reply:
This needs to be `>= 50` because we haven't decremented the budget yet

Reply:
Woops, needed to be `.exists(_ < 50)` because we want this is the failure case